resource-types/typescript-example-website-monitor/src/handlers.ts

```diff
@@ -0,0 +1,316 @@
+            model.endpointRegion = model.endpointRegion || 'US';
+            model.kind = Resource.DEFAULT_MONITOR_KIND;
+            model.status = Resource.DEFAULT_MONITOR_STATUS;
+            model.locations = Resource.DEFAULT_MONITOR_LOCATIONS;
+            model.slaThreshold = Resource.DEFAULT_MONITOR_SLA_THRESHOLD;
+
+            // Update the synthetics monitor by calling the endpoint with its ID.
+            // https://docs.newrelic.com/docs/apis/synthetics-rest-api/monitor-examples/manage-synthetics-monitors-rest-api#update-monitor
+            const apiKey = model.apiKey;
+            const apiEndpoint = ApiEndpoints[model.endpointRegion as EndpointRegions];
+            const response: Response = await fetch(`${apiEndpoint}/v3/monitors/${id}`, {
+                method: 'PUT',
+                headers: { ...Resource.DEFAULT_HEADERS, 'Api-Key': apiKey },
+                body: JSON.stringify({
+                    name,
+                    uri: model.uri,
+                    type: model.kind,
+                    frequency: model.frequency,
+                    status: model.status,
+                    locations: model.locations,
+                    slaThreshold: model.slaThreshold
+                } as Monitor)
+            });
+            await this.checkResponse(response, logger, id);
+            // model.apiKey = null;
+
+            const progress = ProgressEvent.success<ProgressEvent<ResourceModel>>(model);
+            logger.log('progress', progress);
+            return progress;
+        } catch(err) {
+            logger.log(err);
+            if (err instanceof exceptions.BaseHandlerException) {
+                throw err;
+            }
+            return ProgressEvent.failed<ProgressEvent<ResourceModel>>(HandlerErrorCode.InternalFailure, err.message);
+        }
+    }
+
+    /**
+     * CloudFormation invokes this handler when the resource is deleted, either when
+     * the resource is deleted from the stack as part of a stack update operation,
+     * or the stack itself is deleted.
+     *
+     * @param session Current AWS session passed through from caller
+     * @param request The request object for the provisioning request passed to the implementor
+     * @param callbackContext Custom context object to allow the passing through of additional
+     * state or metadata between subsequent retries
+     * @param logger Logger to proxy requests to default publishers
+     */
+    @handlerEvent(Action.Delete)
+    public async delete(
+        session: Optional<SessionProxy>,
+        request: ResourceHandlerRequest<ResourceModel>,
+        callbackContext: CallbackContext,
+        logger: LoggerProxy
+    ): Promise<ProgressEvent> {
+        logger.log('request', request);
+        const model = new ResourceModel(request.desiredResourceState);
+
+        // The Id property, being the primary identifier, cannot be left empty.
+        if (!model.id) {
```

Comment:
Just a general comment here. I don't believe this should have to be done in the code. The service shouldn't be passing something to delete that doesn't have a `Id` since its a `primaryIdentifier`

Reply:
Needed for contract tests